src/fireedge/src/modules/containers/Settings/Menu/ProfileImage.js

```diff
@@ -0,0 +1,143 @@
+import { Avatar, Box, Tooltip, Typography, useTheme } from '@mui/material'
+import { Edit as EditIcon } from 'iconoir-react'
+import { ReactElement, useCallback, useMemo } from 'react'
+
+const styles = ({ palette, typography }) => ({
+  root: css({
+    textAlign: 'center',
+    padding: `${typography.pxToRem(32)} 0 ${typography.pxToRem(16)}`,
+  }),
+  imagePlace: css({
+    width: typography.pxToRem(96),
+    height: typography.pxToRem(96),
+    position: 'relative',
+    display: 'inline-block',
+  }),
+  image: css({
+    width: typography.pxToRem(96),
+    height: typography.pxToRem(96),
+    border: `${typography.pxToRem(2)} solid ${palette.info.dark}`,
+  }),
+  uploadIcon: css({
+    position: 'absolute',
+    bottom: '0px',
+    right: '0px',
+    backgroundColor: palette.secondary.contrastText,
+    border: `${typography.pxToRem(2)} solid ${palette.info.dark}`,
+    borderRadius: '50%',
+    '& > button, & > button:hover, & > button:active': {
+      color: `${palette.info.dark} !important`,
+    },
+  }),
+  userName: css({
+    textAlign: 'center',
+    padding: `0 ${typography.pxToRem(32)} ${typography.pxToRem(16)}`,
+    '& > *': {
+      color: palette.info.main,
+      display: 'inline',
+      textOverflow: 'ellipsis',
+      overflow: 'hidden',
+    },
+  }),
+})
+
+/**
+ * Profile Image Component.
+ *
+ * @returns {ReactElement} ProfileImage component
+ */
+const ProfileImage = () => {
+  const theme = useTheme()
+  const classes = useMemo(() => styles(theme), [theme])
+  const { user } = useAuth()
+  const [updateUser] = UserAPI.useUpdateUserMutation()
+  const { enqueueError } = useGeneralApi()
+
+  const handleImageChange = useCallback(
+    (event) => {
+      const file = event.target.files[0]
+      if (!file) return
+
+      if (file.size > 2 * 1024 ** 2) {
+        enqueueError(T.LimitProfileImage)
+
+        return
+      }
+
+      const reader = new FileReader()
+      reader.onloadend = async () => {
+        const userData = {
+          ...(user?.TEMPLATE?.FIREEDGE || {}),
+          IMAGE_PROFILE: reader.result,
+        }
+
+        const template = jsonToXml({ FIREEDGE: userData })
+        await updateUser({ id: user.ID, template, replace: 1 })
+        event.target.value = null
+      }
+      reader.readAsDataURL(file)
+    },
+    [updateUser, user]
+  )
+
+  const userName = `${user?.NAME || ''}!`
+
+  return (
+    <>
+      <Box className={classes.root}>
+        <Box className={classes.imagePlace}>
+          <Avatar
+            src={user?.TEMPLATE?.FIREEDGE?.IMAGE_PROFILE}
+            className={classes.image}
+          />
+          <Box className={classes.uploadIcon}>
+            <SubmitButton
+              icon={<EditIcon />}
+              importance={STYLE_BUTTONS.IMPORTANCE.SECONDARY}
+              size={STYLE_BUTTONS.SIZE.MEDIUM}
+              type={STYLE_BUTTONS.TYPE.OUTLINED}
+              onClick={() =>
+                document.getElementById('file-upload-input').click()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using Function.prototype.bind and arrows functions as attributes will negatively impact performance in React. Each time the parent is rendered, the function will be re-created and trigger a render of the component causing excessive renders and more memory use.